tests/open_alchemy/helpers/test_peek/test_peek_extension.py

```diff
@@ -646,35 +59,179 @@ def test_mixins(schema, schemas, expected_mixins):
+    WHEN the function called with the schema
+    THEN the expected value is returned.
+    """
+    schema = {f"{prefix}{key}": value for key, value in key_values}
+
+    returned_value = func(schema=schema, schemas={})
+
+    assert returned_value == expected_value
+
+
+@pytest.mark.parametrize(
+    "prefix",
+    [pytest.param(p, id=p) for p in helpers.peek.VALID_PREFIXES],
+)
+@pytest.mark.parametrize(
+    "key_values, func",
+    [
+        pytest.param(
+            [(extension, value)],
+            getattr(helpers.peek, extension.replace("-", "_")),
+            id=f"invalid {extension}",
+        )
+        for extension, value in [
+            ("autoincrement", "True"),
+            ("index", "True"),
+            ("unique", "True"),
+            ("primary-key", "True"),
+            ("tablename", True),
+            ("inherits", 1),
+            ("json", 1),
+            ("backref", True),
+            ("secondary", True),
+            ("uselist", "True"),
+            ("foreign-key", True),
+            ("foreign-key-column", True),
+            ("server-default", True),
+            ("dict-ignore", "True"),
+            ("kwargs", True),
+            ("kwargs", {1: True}),
+            ("kwargs", {1: True, "key": "value"}),
+            ("kwargs", {"key": "value", 1: True}),
+            ("kwargs", {1: True, 2: False}),
+            ("foreign-key-kwargs", True),
+            ("foreign-key-kwargs", {1: True}),
+            ("foreign-key-kwargs", {1: True, "key": "value"}),
+            ("foreign-key-kwargs", {"key": "value", 1: True}),
+            ("foreign-key-kwargs", {1: True, 2: False}),
+        ]
+    ],
+)
 @pytest.mark.helper
-def test_dict_ignore_wrong_type():
+def test_peek_malformed_schema_error(prefix, key_values, func):
     """
-    GIVEN schema with dict_ignore defined as a string
-    WHEN dict_ignore is called with the schema
+    GIVEN schema with autoincrement defined as a string
+    WHEN autoincrement is called with the schema
     THEN MalformedSchemaError is raised.
     """
-    schema = {"x-dict-ignore": "True"}
+    schema = {f"{prefix}{key}": value for key, value in key_values}
 
     with pytest.raises(exceptions.MalformedSchemaError):
-        helpers.peek.dict_ignore(schema=schema, schemas={})
+        func(schema=schema, schemas={})
 
 
 @pytest.mark.parametrize(
-    "schema, expected_dict_ignore",
+    "prefix",
+    [pytest.param(p, id=p) for p in helpers.peek.VALID_PREFIXES],
+)
+@pytest.mark.parametrize(
+    "key_values, func",
     [
-        ({}, None),
-        ({"x-dict-ignore": True}, True),
-        ({"x-dict-ignore": False}, False),
+        pytest.param(
+            [(extension, value)],
+            getattr(helpers.peek, extension.replace("-", "_")),
+            id=f"invalid {extension}",
+        )
+        for extension, value in [
+            ("mixins", "MissingDot"),
+            ("mixins", ".second"),
+            ("mixins", "first."),
+            ("mixins", ".second.third"),
```

Comment:
Are these mixing a duplicate of the above (line 9 to 49)?

Reply:
Nope. These are to test when the mixins are marlfomed, thus will fail. 

The tests line 9 to 49 are testing the valid cases, with some extra attributes (`$ref`, `allOf`).